File: LeetCode/cpp/14.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string> &strs) {
        string res;
        if (strs.size() == 0)
            return res;
        for (int i = 0; i < strs[0].size(); ++i) {
            for (int j = 1; j < strs.size(); ++j) {
                if (i >= strs[j].size() || strs[j][i] != strs[0][i])
                    return res;
            }
            res += strs[0][i];
        }
        return res;
    }
};

File: LeetCode/python/14.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:
            return ""
        pos, res = 0, ""
        for c in strs[0]:
            for i in range(1, len(strs)):
                if pos >= len(strs[i]) or strs[i][pos] != c:
                    return res
            res += c
            pos += 1
        return res
    